src/Functions/identity.h

```diff
@@ -33,6 +33,11 @@ class FunctionIdentityBase : public IFunction
     {
         return arguments.front().column;
     }
+
+    Monotonicity getMonotonicityForRange(const IDataType &, const Field &, const Field &) const override
```

Comment:
We often use `identity` as an optimization-inhibiting function. That's why these changes look controversial.